forte/data/converter/converter.py

```diff
@@ -0,0 +1,343 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import List, Tuple, Any, Optional, Union, Dict
+import numpy as np
+import torch
+
+from forte.common.configuration import Config
+from forte.common import ValidationError
+from forte.data.converter.feature import Feature
+
+logger = logging.getLogger(__name__)
+
+
+class Converter:
+    """
+    This class has the functionality of converting a batch of
+    :class:`forte.data.converter.Feature` to a PyTorch `Tensor`. It will also
+    do the padding for the given batch of :class:`forte.data.converter.Feature`.
+
+    Args:
+        config: An instance of `Dict` or
+            :class:`forte.common.configuration.Config` that provides all
+            configurable options. See :meth:`default_configs` for available
+            options and default values.
+    """
+
+    def __init__(self, config: Union[Dict, Config]):
+        self._config = Config(config,
+                              default_hparams=self.default_configs(),
+                              allow_new_hparam=True)
+
+    @staticmethod
+    def default_configs():
+        """
+        Returns a dictionary of default hyper-parameters.
+
+        .. code-block:: python
+
+            {
+                "to_numpy": True,
+                "to_torch": True
+            }
+
+        Here:
+
+            `"to_numpy"`: bool
+                Whether convert to `numpy.ndarray`.
+                Default is True.
+
+            `"to_torch"`: bool
+                 Whether convert to `torch.tensor`. Default is True.
+
+            .. note::
+                If `need_pad` in :class:`forte.data.converter.Feature`
+                is False and `to_numpy` and `to_torch` is True,
+                it will raise an exception if the target data cannot be
+                converted to `numpy.ndarray` or `torch.tensor`.
+
+        .. note::
+            If `need_pad` in :class:`forte.data.converter.Feature`
+            is True and `to_torch` is True, `to_torch` will overwrite the
+            effect of `to_numpy`.
+        """
+        return {
+            "to_numpy": True,
+            "to_torch": True
+        }
+
+    @property
+    def to_numpy(self) -> bool:
+        return self._config.to_numpy
+
+    @property
+    def to_torch(self) -> bool:
+        return self._config.to_torch
+
+    @property
+    def state(self) -> Dict:
+        return {
+            "to_numpy": self.to_numpy,
+            "to_torch": self.to_torch
+        }
+
+    def load_state(self, state: Dict):
```

Comment:
I guess it is helpful to add tests to cover the state part. You can take a look at the coverage report to see if the test misses the core part.

Reply:
Added